### PR TITLE
Added testing for hipMemGetInfo

### PR DESCRIPTION
**Pull request by ShanoToni**
Testing the ```hipMemGetInfo``` API.
Note that the PR depends on ```HIP_CHECK_THREAD``` and ```HIP_CHECK_ERROR``` from #2664

### CODE REVIEW

File: tests/catch/unit/memory/hipMemGetInfo.cc [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+  auto Malloc1Size = freeMemInit >> 1;
+  // if the allocation is not divisible by the MinAllocation
+  // take into account and add padding
+  fixAllocSize(Malloc1Size);
+  HIP_CHECK(hipMalloc(reinterpret_cast<void**>(&A_mem), Malloc1Size));
+
+  // allocate an extra quarter of free mem
+  auto Malloc2Size = Malloc1Size >> 1;
+  fixAllocSize(Malloc2Size);
+  HIP_CHECK(hipMalloc(reinterpret_cast<void**>(&A_mem), Malloc2Size));
```
Comment (chrispaquot) [inline tests/catch/unit/memory/hipMemGetInfo.cc]: B_mem here instead

File: tests/catch/hipTestMain/config/config_amd_linux.json [hunk 2]
```diff
@@ -1,7 +1,9 @@
 {
     "DisabledTests":
 	[
-	   "Unit_hipStreamPerThread_DeviceReset_1"
+	   "Unit_hipStreamPerThread_DeviceReset_1",
+	   "Unit_hipMemGetInfo_Negative",
+	   "# The prior tests relates to ticket EXSWCPHIPT-61"
```
Comment (chrispaquot) [inline tests/catch/hipTestMain/config/config_amd_linux.json]: Write that info inside the test as a comment or something

File: tests/catch/hipTestMain/config/config_amd_windows.json [hunk 3]
```diff
@@ -66,7 +66,9 @@
 	   "Unit_hipHostMalloc_Coherent",
 	   "Unit_hipHostMalloc_Default",
 	   "Unit_hipStreamCreate_MultistreamBasicFunctionalities",
-	   "Unit_hipEventIpc"
+	   "Unit_hipEventIpc",
+	   "Unit_hipMemGetInfo_Negative",
+	   "# The prior tests relates to ticket EXSWCPHIPT-61"
```
Comment (chrispaquot) [inline tests/catch/hipTestMain/config/config_amd_windows.json]: Write that info inside the test as a comment or something

File: tests/catch/hipTestMain/config/config_amd_linux.json (hunk 2 above)
Comment (ShanoToni) [reply tests/catch/hipTestMain/config/config_amd_linux.json]: Changes to the json files have been removed, added ```HIP_SKIP_TEST``` inside test cases with relevant information

File: tests/catch/hipTestMain/config/config_amd_windows.json (hunk 3 above)
Comment (ShanoToni) [reply tests/catch/hipTestMain/config/config_amd_windows.json]: Changes to the json files have been removed, added ```HIP_SKIP_TEST``` inside test cases with relevant information

File: tests/catch/unit/memory/hipMemGetInfo.cc (hunk 1 above)
Comment (ShanoToni) [reply tests/catch/unit/memory/hipMemGetInfo.cc]: Corrected, thank you!

**Review by chrispaquot — DISMISSED** [review]

**Review by mangupta — DISMISSED** [review]
Reapproving lost due to resolving merge conflicts

**Review by chrispaquot — DISMISSED** [review]

**Review by chrispaquot — DISMISSED** [review]

**Review by chrispaquot — APPROVED** [review]

**Pull request merged**